code/datums/brain_damage/mrat.dm

```diff
@@ -0,0 +1,156 @@
+		new /datum/mrat_type("Slime Puppy", 'icons/mob/pets.dmi', "slime_puppy", "sound/machines/uplinkpurchase.ogg"),
+		new /datum/mrat_type("Chick", 'icons/mob/animal.dmi', "chick", "sound/effects/mousesqueek.ogg"),
+		new /datum/mrat_type("Mothroach", 'icons/mob/animal.dmi', "mothroach", "sound/voice/moth/scream_moth.ogg", type_volume=50),
+		new /datum/mrat_type("Bee", 'icons/mob/animal.dmi', "bee_big", "sound/voice/moth/scream_moth.ogg", type_volume=50),
+		new /datum/mrat_type("Butterfly", 'icons/mob/animal.dmi', "butterfly", "sound/voice/moth/scream_moth.ogg", type_color="#1ABC9C", type_volume=50),
+		new /datum/mrat_type("Hologram", 'icons/mob/ai.dmi', "default", "sound/machines/ping.ogg", type_volume=50),
+		new /datum/mrat_type("Spaceman", 'icons/mob/animal.dmi', "old", "sound/machines/buzz-sigh.ogg", type_volume=50)
+	)
+
+/mob/camera/imaginary_friend/mrat/proc/update_available_icons()
+	icons_available = list()
+
+	for(var/datum/mrat_type/T in mrat_types)
+		icons_available += list("[T.name]" = image(icon = T.icon, icon_state = T.icon_state))
+
+/mob/camera/imaginary_friend/mrat/proc/Costume()
+	update_available_icons()
+	if(icons_available)
+		var/selection = show_radial_menu(src, src, icons_available, radius = 38)
+		if(!selection)
+			return
+
+		for(var/datum/mrat_type/T in mrat_types)
+			if(T.name == selection)
+				current_costume = T
+				human_image = image(icon = T.icon, icon_state = T.icon_state)
+				color = T.color
+				Show()
+				return
+
+/mob/camera/imaginary_friend/mrat/proc/PickName()
+	var/picked_name = sanitize_name(stripped_input(src, "Enter your mentor rat's name", "Rat Name", "Mentor Rat", MAX_NAME_LEN - 3 - length(key)))
+	if(!picked_name || picked_name == "")
+		picked_name = "Mentor Rat"
+	log_game("[key_name(src)] has set \"[picked_name]\" as their mentor rat's name for [key_name(owner)]")
+	name = "[picked_name] ([key])"
+
+/mob/camera/imaginary_friend/mrat/friend_talk()
+	. = ..()
+	if(!current_costume || !istype(current_costume))
+		return
+	SEND_SOUND(owner, sound(current_costume.sound, volume=current_costume.volume))
+	SEND_SOUND(src, sound(current_costume.sound, volume=current_costume.volume))
+
+/mob/camera/imaginary_friend/mrat/greet()
+	to_chat(src, "<span class='notice'><b>You are the mentor rat of [owner]!</b></span>")
+	to_chat(src, "<span class='notice'>Do not give [owner] any OOC information from your time as a ghost.</span>")
+	to_chat(src, "<span class='notice'>Your job is to answer [owner]'s question(s) and you are given this form to assist in that.</span>")
+	to_chat(src, "<span class='notice'>Don't be stupid with this or you will face the consequences.</span>")
+
+/mob/camera/imaginary_friend/mrat/Initialize(mapload, _trauma)
+	. = ..()
+	costume = new
+	costume.Grant(src)
+	leave = new
+	leave.Grant(src)
+	grant_all_languages()
+
+/mob/camera/imaginary_friend/mrat/setup_friend()
+	human_image = null
```

Comment:
Where does the image get added to the requester client?

Reply:
parent imaginary friend datum